### ISSUE TITLE
Yolo transfer learning

### ISSUE THREAD

**Issue by rammanoharvs** [ultralytics/ultralytics#15997]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,
Basically I have a custom  dataset which I need to train on YOLO8s-seg.pt. I do have the images and annotation.

Issue is that once I train the yolo8s-seg with custom dataset, yolo is forgetting the coco classes .  So for this I read that we can do transfer learning so I started with a small learning rate which Adam (0.0001) and froze all but the last layer, then the last 2 layers but still transfer learning is really not happening

Basically code is
YOLO("yolov8s-seg.pt")
for k, v in model.model.named_parameters(): 
     v.requires_grad = False  # Freeze all layers 
#above freezes all layers but then i tried freezing some also

The above freezing means that there should not be anything learnt from the custom dataset and it should still recognize the coco classes of course not custom dataset onces. YAML contains all the coco classes + one of my class crane which is marked in new dataset.

What is happenning is that even with all freezing (tried with freeze=21 as well parameter) , coco classes are forgotten and new dataset class is predicted old ones no
I had taken the base model on that folder and checked before that is it predicting the coco classes and initially it was predicting but after one training with custom dataset it is not predicting coco classes

Any directions more on this one? Of course this is YOLOv8

**Comment by Y-T-G** [comment]
It will eventually always forget the COCO classes. Transfer learning can't prevent this. This is known as catastrophic forgetting. As long as you're not training on a dataset that has all the classes that you want to keep, it will always start to forget those classes.

**Comment by rammanoharvs** [comment]
So i have 11k images and let us say i mark all coco classes available there as well as my classes ( of course not all coco classes are available in the dataset but whatever is there) then it will help the model remember coco classes or it will still remember only what we marked or some lighting conditions or some shapes which it got trained with Coco classes also?

**Comment by Y-T-G** [comment]
It will remember the features of the classes from your dataset, so it may lose some accuracy and generalizability because the actual COCO dataset has 118k images with lots of variations which may not be present in your dataset for those classes.

**Comment by rammanoharvs** [comment]
Now the question arises on what would be the use of using pre trained model if whatever is pretrained does not remain if you try to add on some more classes and images? So if we really need to train then we need retrain coco and ours?

I would say that is against the meaning of a pretrained model. Is this true?

**Comment by Y-T-G** [comment]
A pretrained model is used for transfer learning for faster convergence and to reduce overfitting.

Transfer learning isn't about retaining the old classes, but rather "seeding" the model during training with prior information that it has learnt from being trained on a much larger model. This helps the model generalize better as opposed to training from scratch.

**Comment by rammanoharvs** [comment]
So that generalization still remains if we do transfer learning? My results show that even with yaml being retained and added no detection really happens. New classes does happen . What is the use of this generalization?  How is it different from scratch?

**Comment by Y-T-G** [comment]
Generalizability means the model works better on unseen data. So your model may perform well on your dataset, but when you deploy it, it performs worse. This would be because the model doesn't generalize well.

Training from scratch usually has worse generalizability because the model only exclusively learns features from your dataset. So the smaller and less diverse your dataset is, the more likely it will overfit and not generalize well. But a model trained on a large dataset like COCO has learnt the general features from a variety of images which is beneficial for transfer learning.

> I mean if i train something and if it only retains that then i would call it training from scratch

That's simply how deep learning works. You can read more about catastrophic forgetting online.

**Comment by rammanoharvs** [comment]
Thanks for the answers so I would say if i give a mix of my new classes and old classes it would ultimately have learnt and generalized this as well as coco datasets features. So while prediction it would work out. Am I correct?

**Comment by Y-T-G** [comment]
It will retain the classes. But will it be as good as the original, that depends. You'll have to train and see how much it forgot.

**Comment by rammanoharvs** [comment]
Hopefully it will be. The plan is to put some images from coco dataset and also our dataset annotate all classes possible. This way model remembers. Hopefully it works

**Comment by rammanoharvs** [comment]
Anyway thanks for all the replies
